ext/phar/phar.c

```diff
@@ -2488,42 +2488,48 @@ static int phar_flush_clean_deleted_apply(zval *zv) /* {{{ */
 
 #include "stub.h" /* Generated phar_get_stub() function from makestub.php script */
 
-zend_string *phar_create_default_stub(const char *index_php, const char *web_index, char **error) /* {{{ */
+zend_string *phar_create_default_stub(const zend_string *php_index_str, const zend_string *web_index_str, char **error) /* {{{ */
```

Comment:
I think the old code in this function is clearer.
High chance that the compiler pre-computes the string length anyway in the !index and !web_index branches, so no need to optimize that.
The changes to "P" instead of "S" in ZPP should be kept though

Reply:
> Should I also keep the change where it returns NULL in case the path is over 400 even if error == NULL?

Yes

> Also the other motivation was to get rid of a strlen call that is not needed

Right, but I think the compiler is smart enough to avoid it.


Anyway, reading this again it's clearer to me now, so you may keep it anyway.